### PR TITLE
[java][spring] added springdoc support for service metadata and security

### PR DESCRIPTION
**Pull request by bluenick2k17**
- Added the auto-generated OpenAPI 3 support for service metadata and security, using springdoc.
  - Changes made to `SpringCodegen` to add the new documentation config file (`SpringDocConfiguration.java`) to the list of auto-generated files
  - Added new template file: `springdocDocumentationProvider.mustache` for the Spring generator which builds `SpringDocConfiguration.java`
  - This can be validated by generating a java spring API using the default documentation provider, and examining `target/generated-sources/openapi/src/main/java/org/openapitools/configuration/SpringDocConfiguration.java` in the resulting API.
  - Updated spring samples in this PR can also be checked to see some examples of the new springdoc configuration files.
- `CodegenSecurityTest` was updated to test the new addition of the SpringDoc configuration, and test was cleaned up to reduce duplication when adding the additional documentation provider test.
- Fixed a piece of documentation and the connected code in `SpringCodegen` which still used Springfox verbiage, even though the default documentation provider is springdoc.

Some notes on the implementation:

- Did not remove Springfox, since it's marked for removal in 7.0.0: https://github.com/OpenAPITools/openapi-generator/issues/11553
- Current pull request supports basic and API key fully with the security info. Does not yet support Oauth2 completely since it would require a bunch more work into the template. Would affect the ability to use auth on the swagger-ui with Oauth2
- `CodegenSecurity` doesn't support a description field yet for the auth method, so as of now that support remains the same

Closes https://github.com/OpenAPITools/openapi-generator/issues/12220

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02)

### CODE REVIEW

**Review by borsch — APPROVED** [review]

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/java/spring/SpringCodegenTest.java [hunk 1]
```diff
@@ -877,13 +892,13 @@ public void testConfigFileGeneration() {
             tmpFile = s.getTemplateFile();
             desFile = s.getDestinationFilename();
```
Comment (welshm) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/java/spring/SpringCodegenTest.java:893]: Is there any value in testing the contents of the file?

Can always be added later if needed to validate any future bug fixes

**Review by welshm — APPROVED** [review]
Thanks for adding this!

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/java/spring/SpringCodegenTest.java (hunk 1 above)
Comment (welshm) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/java/spring/SpringCodegenTest.java:893]: Yeah - just comparing to the test above which actually runs the generator and then validates the file content 

```
        assertFileContains(Paths.get(outputPath + "/src/main/java/org/openapitools/api/GetApi.java"),
            "@RequestParam(value = \"testParameter1\", required = false, defaultValue = \"BAR\")",
            "@RequestParam(value = \"TestParameter2\", required = false, defaultValue = \"BAR\")");
```

But that might be overkill to do that for this test, at this point anyway.

**Review by welshm — APPROVED** [review]

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/java/spring/SpringCodegenTest.java (hunk 1 above)
Comment (bluenick2k17) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/java/spring/SpringCodegenTest.java:893]: I saw your example below - thanks for showing it! I didn't spot that when I was thinking about it. I think I agree it's a little overkill for this feature, but definitely will keep this in mind for the future. As long as you have no existing concerns on this thread- I'll let you resolve the conversation when you get the chance :). I appreciate the discussion!

**Review by cachescrubber — APPROVED** [review]

**Comment by wing328** [comment]
Thanks again for the PR, which has been included in the v6.0.0 release: https://twitter.com/oas_generator/status/1529669733934841856

**Pull request merged**